src/lib.rs

```diff
@@ -387,31 +492,33 @@ impl PerFilePendingFixes {
 type RuleIndex = usize;
 type RuleListenerIndex = usize;
 
-struct AggregatedQueries<'resolved_rules> {
+struct AggregatedQueries<'a> {
     pattern_index_lookup: Vec<(RuleIndex, RuleListenerIndex)>,
     #[allow(dead_code)]
     query: Query,
     query_text: String,
-    resolved_rules: &'resolved_rules [ResolvedRule<'resolved_rules>],
+    instantiated_rules: &'a [InstantiatedRule],
 }
 
-impl<'resolved_rules> AggregatedQueries<'resolved_rules> {
-    pub fn new(
-        resolved_rules: &'resolved_rules [ResolvedRule<'resolved_rules>],
-        config: &Config,
-    ) -> Self {
+impl<'a> AggregatedQueries<'a> {
+    pub fn new(instantiated_rules: &'a [InstantiatedRule], language: SupportedLanguage) -> Self {
         let mut pattern_index_lookup: Vec<(RuleIndex, RuleListenerIndex)> = Default::default();
         let mut aggregated_query_text = String::new();
-        for (rule_index, resolved_rule) in resolved_rules.into_iter().enumerate() {
-            for (rule_listener_index, rule_listener) in resolved_rule.listeners.iter().enumerate() {
-                for _ in 0..rule_listener.query.pattern_count() {
+        let language = language.language();
+        for (rule_index, instantiated_rule) in instantiated_rules.into_iter().enumerate() {
+            for (rule_listener_index, rule_listener_query) in instantiated_rule
+                .rule
+                .listener_queries()
+                .iter()
+                .map(|rule_listener_query| rule_listener_query.resolve(language))
```

Comment:
So there's still a concept of a rule listener "resolving" but it's just the "query" part (vs the "callback" part) that's being resolved (from query text + optional capture name -> parsed query + capture index)